--- intel-vaapi-driver-2.3.0/configure.ac	2018-12-09 21:17:18.000000000 -0500
+++ intel-vaapi-driver-2.3.0.old/configure.ac	2020-05-18 18:13:42.747683262 -0400
@@ -92,7 +92,7 @@
 if test "X$CC-cc" != "X"; then
     AC_MSG_CHECKING([whether ${CC-cc} accepts -fstack-protector])
     ssp_old_cflags="$CFLAGS"
-    CFLAGS="$CFLAGS -fstack-protector"
+    CFLAGS="$CFLAGS -fstack-protector -fcommon"
     AC_LINK_IFELSE([AC_LANG_PROGRAM([[]], [[alloca(100);]])], [], [ssp_cc=no])
     AC_MSG_RESULT([$ssp_cc])
     if test "X$ssp_cc" = "Xno"; then
