--- a/third_party/WebKit/Source/platform/graphics/gpu/SharedGpuContext.h  2017-05-15 17:44:50.728855047 -0400
+++ b/third_party/WebKit/Source/platform/graphics/gpu/SharedGpuContext.h  2017-05-15 17:38:49.517846462 -0400
@@ -5,6 +5,7 @@
 #include "platform/PlatformExport.h"
 #include "platform/wtf/ThreadSpecific.h"

+#include <functional>
 #include <memory>

 namespace gpu {

diff --git a/third_party/WebKit/Source/platform/wtf/LinkedHashSet.h b/third_party/WebKit/Source/platform/wtf/LinkedHashSet.h
index e85c72f..6f94cd6 100644
--- a/third_party/WebKit/Source/platform/wtf/LinkedHashSet.h
+++ b/third_party/WebKit/Source/platform/wtf/LinkedHashSet.h
@@ -685,6 +685,8 @@
   return *this;
 }

+inline void SwapAnchor(LinkedHashSetNodeBase& a, LinkedHashSetNodeBase& b);
+
 template <typename T, typename U, typename V, typename W>
 inline void LinkedHashSet<T, U, V, W>::Swap(LinkedHashSet& other) {
   impl_.Swap(other.impl_);
