diff -Naur backup/tmp/config.def.h work/config.def.h
--- backup/tmp/config.def.h	2017-03-21 23:52:19.944435889 -0400
+++ work/config.def.h	2017-03-21 23:51:57.386435353 -0400
@@ -150,20 +150,10 @@
  */
 unsigned int defaultattr = 11;
 
-/*
- * Internal mouse shortcuts.
- * Beware that overloading Button1 will disable the selection.
- */
-MouseShortcut mshortcuts[] = {
-	/* button               mask            string */
-	{ Button4,              XK_NO_MOD,      "\031" },
-	{ Button5,              XK_NO_MOD,      "\005" },
-};
-
 MouseKey mkeys[] = {
-	/* button               mask            function        argument */
-	{ Button4,              ShiftMask,      kscrollup,      {.i =  1} },
-	{ Button5,              ShiftMask,      kscrolldown,    {.i =  1} },
+	/* button               function        argument */
+	{ Button4,              kscrollup,      {.i =  1} },
+	{ Button5,              kscrolldown,    {.i =  1} },
 };
 
 /* Internal keyboard shortcuts. */
diff -Naur backup/tmp/st.c work/st.c
--- backup/tmp/st.c	2017-03-21 23:52:19.944435889 -0400
+++ work/st.c	2017-03-21 23:51:57.386435353 -0400
@@ -236,7 +236,6 @@
 
 /* config.h array lengths */
 size_t colornamelen = LEN(colorname);
-size_t mshortcutslen = LEN(mshortcuts);
 size_t mkeyslen = LEN(mkeys);
 size_t shortcutslen = LEN(shortcuts);
 size_t selmaskslen = LEN(selmasks);
diff -Naur backup/tmp/st.h work/st.h
--- backup/tmp/st.h	2017-03-21 23:52:19.945435889 -0400
+++ work/st.h	2017-03-21 23:51:57.386435353 -0400
@@ -145,12 +145,6 @@
 } TermWindow;
 
 typedef struct {
-	uint b;
-	uint mask;
-	char *s;
-} MouseShortcut;
-
-typedef struct {
 	int mode;
 	int type;
 	int snap;
@@ -182,7 +176,6 @@
 
 typedef struct {
 	uint b;
-	uint mask;
 	void (*func)(const Arg *);
 	const Arg arg;
 } MouseKey;
@@ -277,8 +270,6 @@
 extern unsigned int defaultattr;
 extern MouseKey mkeys[];
 extern size_t mkeyslen;
-extern MouseShortcut mshortcuts[];
-extern size_t mshortcutslen;
 extern Shortcut shortcuts[];
 extern size_t shortcutslen;
 extern uint forceselmod;
diff -Naur backup/tmp/x.c work/x.c
--- backup/tmp/x.c	2017-03-21 23:52:19.945435889 -0400
+++ work/x.c	2017-03-21 23:51:57.386435353 -0400
@@ -247,7 +247,6 @@
 bpress(XEvent *e)
 {
 	struct timespec now;
-	MouseShortcut *ms;
     MouseKey *mk;
 
 	if (IS_SET(MODE_MOUSE) && !(e->xbutton.state & forceselmod)) {
@@ -256,20 +255,11 @@
 	}
 
     for (mk = mkeys; mk < mkeys + mkeyslen; mk++) {
-		if (e->xbutton.button == mk->b
-				&& match(mk->mask, e->xbutton.state)) {
+		if (e->xbutton.button == mk->b) {
 			mk->func(&mk->arg);
 			return;
 		}
 	}
-
-	for (ms = mshortcuts; ms < mshortcuts + mshortcutslen; ms++) {
-		if (e->xbutton.button == ms->b
-				&& match(ms->mask, e->xbutton.state)) {
-			ttysend(ms->s, strlen(ms->s));
-			return;
-		}
-	}
 
 	if (e->xbutton.button == Button1) {
 		clock_gettime(CLOCK_MONOTONIC, &now);
