diff -aur dwm-tmp/dwm.c dwm/dwm.c
--- a/dwm.c	2017-05-24 21:44:20.054378944 -0400
+++ b/dwm.c	2017-06-24 12:30:22.439194880 -0400
@@ -750,8 +750,10 @@

 	if ((w = m->ww - sw - x) > bh) {
 		if (m->sel) {
+            char ascii[256];
+            strip_utf(m->sel->name, ascii);
 			drw_setscheme(drw, scheme[m == selmon ? SchemeSel : SchemeNorm]);
-			drw_text(drw, x, 0, w, bh, lrpad / 2, m->sel->name, 0);
+			drw_text(drw, x, 0, w, bh, lrpad / 2, ascii, 0);
 			if (m->sel->isfloating)
 				drw_rect(drw, x + boxs, boxs, boxw, boxw, m->sel->isfixed, 0);
 		} else {
diff -aur dwm-tmp/util.c dwm/util.c
--- a/util.c	2017-05-24 21:40:16.331373152 -0400
+++ b/util.c	2017-06-24 12:34:35.554200896 -0400
@@ -33,3 +33,14 @@

 	exit(1);
 }
+
+void
+strip_utf(char* buf, char* mt) {
+    for (char* text = buf; *text != '\0'; text++) {
+        if (*text >= 0 && *text <= 127) {
+           *mt = *text;
+           mt++;
+        }
+    }
+    *mt = '\0';
+}
diff -aur dwm-tmp/util.h dwm/util.h
--- a/util.h	2017-05-24 21:40:16.331373152 -0400
+++ b/util.h	2017-06-23 23:33:28.458087031 -0400
@@ -6,3 +6,4 @@

 void die(const char *fmt, ...);
 void *ecalloc(size_t nmemb, size_t size);
+void strip_utf(char* buf, char* mt);
